UnityProject/Assets/Scripts/NPC/AI/MobExplore.cs

```diff
@@ -79,6 +108,80 @@ public void OnSpawnServer(SpawnInfo info)
 			eatFoodSound = CommonSounds.Instance.EatFood;
 		}
 
+		//https://www.geeksforgeeks.org/bresenhams-circle-drawing-algorithm/
+		// Function for circle-generation
+		// using Bresenham's algorithm
+		private void CircleBres(int xc, int yc, int r)
+		{
+			int x = 0, y = r;
+			int d = 3 - 2 * r;
+			DrawCircle(xc, yc, x, y);
+			while (y >= x)
+			{
+				// for each pixel we will
+				// draw all eight pixels
+
+				x++;
+
+				// check for decision parameter
+				// and correspondingly
+				// update d, x, y
+				if (d > 0)
+				{
+					y--;
+					d = d + 4 * (x - y) + 10;
+				}
+				else
+					d = d + 4 * x + 6;
+
+				DrawCircle(xc, yc, x, y);
+			}
+		}
+
+		// Function to put Locations
+		// at subsequence points
+		private void DrawCircle(int xc, int yc, int x, int y)
+		{
+			visionCirclePerimeter.Add(new Vector2Int(xc + x, yc + y));
+			visionCirclePerimeter.Add(new Vector2Int(xc - x, yc + y));
+			visionCirclePerimeter.Add(new Vector2Int(xc + x, yc - y));
+			visionCirclePerimeter.Add(new Vector2Int(xc - x, yc - y));
+			visionCirclePerimeter.Add(new Vector2Int(xc + y, yc + x));
+			visionCirclePerimeter.Add(new Vector2Int(xc - y, yc + x));
+			visionCirclePerimeter.Add(new Vector2Int(xc + y, yc - x));
+			visionCirclePerimeter.Add(new Vector2Int(xc - y, yc - x));
+		}
+
+		//https://rosettacode.org/wiki/Bitmap/Bresenham%27s_line_algorithm#C.23
+		private void FillCircle(int xc, int yc, int z)
```

Comment:
Instead of doing this manually, you can use Physics2D.OverlapCircle and the layer masks to check for.